spec/rubocop/cop/style/redundant_regexp_argument_spec.rb

```diff
@@ -47,6 +47,17 @@
     RUBY
   end
 
+  it 'registers an offense and corrects when using single quote character' do
+    expect_offense(<<~RUBY)
+      str.gsub(/'/)
+               ^^^ Use string `'\\''` as argument instead of regexp `/'/`.
+    RUBY
+
+    expect_correction(<<~RUBY)
+      str.gsub('\\'')
+    RUBY
```

Comment:
I think it'd be better to use `'RUBY'` here, e.g.

```suggestion
    expect_correction(<<~'RUBY')
      str.gsub('\'')
    RUBY
```

Reply:
I think we need a cop for this. ;-)